docs/learn/response_models.md

```diff
@@ -240,13 +281,80 @@ except ValidationError as e:
-- **Start Simple**: Begin with basic types and gradually increase complexity as needed.
-- **Handle Errors Gracefully**: Implement proper error handling and consider using retry mechanisms.
-- **Leverage JSON Mode**: When possible, use `json_mode=True` for better type support and consistency.
-- **Test Thoroughly**: Validate your Response Models across different inputs and edge cases.
+## Best Practices
 
-By leveraging Response Models effectively, you can create more robust, type-safe, and maintainable LLM-powered applications with Mirascope.
+Here are some best practices to follow when using Response Models:
+
+1. **Use Clear Field Names**: Choose descriptive names for your model fields to guide the LLM's output.
+   ```python
+   class BookReview(BaseModel):
+       book_title: str
+       author_name: str
+       rating: int
+       review_text: str
+   ```
+
+2. **Provide Field Descriptions**: Use Pydantic's `Field` with descriptions to give the LLM more context.
+   ```python
+   class MovieRecommendation(BaseModel):
+       title: str = Field(..., description="The full title of the movie")
+       director: str = Field(..., description="The name of the movie's director")
+       year: int = Field(..., description="The year the movie was released")
+       genre: str = Field(..., description="The primary genre of the movie")
+   ```
+
+3. **Start Simple**: Begin with basic types and gradually increase complexity as needed.
+   ```python
+   # Start with this:
+   class SimpleRecipe(BaseModel):
+       name: str
+       ingredients: list[str]
+
+   # Then evolve to this:
+   class DetailedRecipe(BaseModel):
+       name: str
+       ingredients: list[Ingredient]
+       preparation_time: int
+       cooking_time: int
+       instructions: list[str]
+   ```
+4. **Handle Errors Gracefully**: Implement proper error handling and consider using retry mechanisms.
+   ```python
+   from tenacity import retry, stop_after_attempt, wait_fixed
+
+   @retry(stop=stop_after_attempt(3), wait=wait_fixed(2))
+   @openai.call("gpt-4o-mini", response_model=Recipe)
+   @prompt_template()
+   def get_recipe(dish: str) -> str:
+       return f"Provide a recipe for {dish}"
+
+   try:
+       recipe = get_recipe("spaghetti carbonara")
+       print(recipe)
+   except Exception as e:
+       print(f"Failed to get recipe after retries: {e}")
+   ```
+ 
+5. **Leverage JSON Mode**: When possible, use `json_mode=True` for better type support and consistency.
+   ```python
+   @openai.call("gpt-4o-mini", response_model=ComplexDataStructure, json_mode=True)
+   @prompt_template()
+   def get_complex_data(query: str) -> str:
+       return f"Generate complex data for {query}"
+   ```
+
+6. **Test Thoroughly**: Validate your Response Models across different inputs and edge cases.
+   ```python
+   import pytest
+
+   @pytest.mark.parametrize("genre", ["sci-fi", "romance", "mystery", "non-fiction"])
```

Comment:
hmm, this test feels more like it's testing mirascope than the desired behavior (since we should expect if it doesn't fail that we will in fact get a book back)